### PR TITLE
Led map

### PR DESCRIPTION
**Pull request by merbanan**
This is a WIP implementation of led map support. Only compile tested for now.

This changes the phy led dts syntax. But I think this describes very well what function is muxed onto a specific pin.

With a matching switch phy driver there would be total freedom with regards to routing of the phy led signals.

### CODE REVIEW

**Comment by merbanan** [comment]
I forgot to add that it is not known if the mapping is correct. The led map register has 5 slots and there is only 4 switch port phys.

File: target/linux/airoha/dts/en7581.dtsi [hunk 1]
```diff
@@ -223,6 +223,7 @@
 		pio: pinctrl {
 			compatible = "airoha,en7581-pinctrl";
 			reg = <0x0 0x1fa20214 0x0 0x30>,
+			      <0x0 0x1fa2027C 0x0 0x8>,
```
Comment (LorenzoBianconi) [inline target/linux/airoha/dts/en7581.dtsi:226]: @merbanan no capital letter

File: target/linux/airoha/files/drivers/pinctrl/pinctrl-airoha.c [hunk 2]
```diff
@@ -96,6 +96,71 @@
 #define REG_FORCE_GPIO32_EN			0x18
 #define REG_INIC_MDIO_SLV_MOD			0x1c
 
+/* LED MAP */
+#define REG_LAN0_LED0_MAPPING			0x68
+#define LAN4_LED0_MAPPING_MASK 			GENMASK(18, 16)
+#define PORT4_LED0_PHY0_LED0_MAP		(0 << 16)
```
Comment (LorenzoBianconi) [inline target/linux/airoha/files/drivers/pinctrl/pinctrl-airoha.c:102]: @merbanan please use BIT macro

File: target/linux/airoha/files/drivers/pinctrl/pinctrl-airoha.c [hunk 3]
```diff
@@ -96,6 +96,71 @@
...
+#define PORT1_LED0_PHY3_LED0_MAP		(3 << 4)
+#define PORT1_LED0_PHY4_LED0_MAP		(4 << 4)
+#define LAN0_LED0_MAPPING_MASK 			GENMASK(2, 0)
+#define PORT0_LED0_PHY0_LED0_MAP		(0 << 0)
+#define PORT0_LED0_PHY1_LED0_MAP		(1 << 0)
+#define PORT0_LED0_PHY2_LED0_MAP		(2 << 0)
+#define PORT0_LED0_PHY3_LED0_MAP		(3 << 0)
+#define PORT0_LED0_PHY4_LED0_MAP		(4 << 0)
+
+#define REG_LAN0_LED1_MAPPING			0x6C
```
Comment (LorenzoBianconi) [inline target/linux/airoha/files/drivers/pinctrl/pinctrl-airoha.c:132]: @merbanan no capital letter

File: target/linux/airoha/files/drivers/pinctrl/pinctrl-airoha.c [hunk 4]
```diff
@@ -623,6 +685,30 @@ static const char *const serdes_groups[] = { "ssusbp0_serdes",
 					     "xpon_serdes",
 					     "ssusbp1_serdes" };
 
+static const char *const phy1_led0_groups[] = { "gpio33", "gpio34",
+					        "gpio35",  "gpio42" };
```
Comment (LorenzoBianconi) [inline target/linux/airoha/files/drivers/pinctrl/pinctrl-airoha.c:689]: @merbanan what is the difference of having a single led group here?

Comment (merbanan) [reply target/linux/airoha/files/drivers/pinctrl/pinctrl-airoha.c:689]: Per my understanding of the code macros the groups need to have these names to match the function declaration. The led map register makes it possible to mux the phy[1-4] led0 signals to any of the led0 gpio pins. And the same for the led1 gpio pins. This equals to 8 functions and thus we can not have only one led group.

**Pull request closed without merging**